Advanced R course exercises
===========================

Exercise 1
----------

Read in the data we were given
```{r}
setwd("o:/Training/Advanced R/Advanced R Data/")
read.delim("methylation.txt") -> methylation
read.delim("expression.txt") -> expression
```

Remove any lines from the methylation data where the promoter methylation equals -1 (ie not actually measured), and see what effect this has on the number of rows of data we have

```{r}
nrow(methylation)
methylation[methylation$Promoter_meth != -1,] -> methylation
nrow(methylation)
```

Deduplicate the expression data based on probe names.  We're going to leave one instance of each duplicated name behind.

```{r}
expression[!duplicated(expression$Probe),] -> expression
```

Find out how many olfactory receptors we have in our expression data

Exercise 2
----------

```{r}
sum(grepl("Olfr",expression$Probe))
```

Change the chromosome names from 1,2,3 etc, to chr1,chr2,chr3 etc.

```{r}
expression$Chromosome <- paste("chr",expression$Chromosome,sep="")
```

Find out which probes are present in the methylation data, but not the expression data

```{r}
methylation$Probe[!(methylation$Probe %in% expression$Probe)]
```



Merge the expression and methylation datasets together.  The only column name in common is Probe, which is what we want to merge on anyway.

```{r}
merge(expression,methylation) -> merged.data
```

Exercise 3
----------

Loop over the expression/methylation columns to calculate the range of values in each.

```{r}
apply(merged.data[,6:8],2,range)
```

Calculate the mean gene body methylation for each chromosome, then plot this as a barplot

```{r fig.width=15}
tapply(merged.data$Gene_body_meth,merged.data$Chromosome,mean)
barplot(tapply(merged.data$Gene_body_meth,merged.data$Chromosome,mean))
```

Exercise 4
----------

Set up a function to calculate the SEM

```{r}
sem <- function (x,absent=NA) {
  
  if (sum(is.na(x)) > 0) return (absent)
  
  return (sd(x)/sqrt(length(x)))
}
```

Calculate the gene body methylation SEM for each chromosome.

```{r}
tapply(merged.data$Gene_body_meth,merged.data$Chromosome,sem)
```

Exercise 5
----------

Install the vioplot library and use it to draw violin plots for the 3 datasets we have.

```{r message=FALSE,fig.width=10,results='hide'}
library(vioplot)
par(mfrow=c(1,3))
sapply(colnames(merged.data)[6:8],function(x) {
  vioplot(merged.data[[x]])
  title(x)
}
)
```

